core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -270,6 +270,10 @@ && isConstant(node.getArguments().get(0))) {
                         // date_trunc(day, a_date) is a no-op
                         return treeRewriter.rewrite(argument, context);
                     }
+                    if (argumentType instanceof TimestampType) {
+                        // prefer `CAST(CAST(x as DATE) AS timestamp)` to `date_trunc('day', x)`, see e.g. UnwrapCastInComparison
```

Comment:
is that still preferable when it comes to CPU cycles when we do not end up pushing down the expression?

Reply:
Good question. This i don't know.
Do you think I should not do this?

Reply:
Well - I think there may be a tradoff. But I do not know the answer. My guts say it is better to be able to do pushdown that have slight (if any) performance degradataion. To know if there really is any negative impact of that we would need some micro benchmarking - maybe we have alreayd IDK.

Reply:
> Well - I think there may be a tradoff. But I do not know the answer. My guts say it is better to be able to do pushdown that have slight (if any) performance degradataion.

I am convinced about that as well.

I can try to handle this case is the comparison context only.

Reply:
> I can try to handle this case is the comparison context only.

Done. This is now removed from here, taken care of by `UnwrapDateTruncInComparison`.